modules/user-access/pages/acl-management.adoc

```diff
@@ -0,0 +1,309 @@
+
+If a query does not have a current owner due to upgrades from an older version of TigerGraph, then
+
+==== Syntax
+[.wrap,gsql]
+----
+CHANGE OWNER OF QUERY <query_name> TO <username> [SECURED BY <acl_password>]
+----
+
+==== Required privilege
+* None if the query is owned by the current user
+* `WRITE_ROLE` if the query is owned by another user
+
+==== Procedure
+To change the owner of a query, run the following command and replace `owner_password` with the ACL password of the query owner.
+[.wrap,gsql]
+----
+GSQL > CHANGE OWNER of QUERY my_query TO user1 SECURED BY "owner_acl_password"
+----
+
+[#_grant_acl_privilege_to_a_role]
+=== Grant ACL privilege to a role
+The owner and *only* the owner of a query can grant ACL privileges on the query to specified roles.
+To grant an ACL privilege, the query owner must have the corresponding RBAC privilege.
+
+==== Syntax
+[.wrap,gsql]
+----
+GRANT ACL PRIVILEGE ( READ | EXECUTE ) ON QUERY TO ROLE (<role1>, <role2> ...) [SECURED BY <owner_acl_password>]
+----
+
+==== Required privilege
+* `READ_QUERY` for granting `READ` privilege on the query.
+* `READ_DATA` for granting `EXECUTE` privilege on queries that do not update the graph.
+* `READ_DATA` and `WRITE_DATA` for granting `EXECUTE` privilege on queries that update the graph.
+
+==== Procedure
+To grant `READ` privilege for a query `example_query` to a role `example_role`, run the following command.
+To grant `EXECUTE` privilege instead, replace `READ` with `EXECUTE`:
+
+[.wrap,gsql]
+----
+GSQL > GRANT ACL PRIVILEGE READ ON example_query TO ROLE example_role
+----
+
+
+=== Revoke ACL privilege from a role
+The owner of a query can revoke ACL privileges on the query from a role.
+If you revoke an ACL privilege on the query from all roles, then the ACL entry status for that privilege becomes unspecified.
+When the ACL entry status for a privilege is unspecified, RBAC governs access for the corresponding operation.
+
+==== Syntax
+[.wrap,gsql]
+----
+REVOKE ACL PRIVILEGE ( READ | EXECUTE ) ON QUERY <query_name> FROM ( <role1>, <role2>, ... ) | ALL ) [SECURED BY <owner_acl_password>]
+----
+
+==== Required privilege
+None. Only query owner can run the command.
+If the query owner set their ACL password, running the command requires entering the ACL password.
+
+==== Procedure
+To revoke ACL `READ` privilege on `example_query` from `role1`, run the following command and replace `owner_acl_password` with the ACL password of the owner.
+If you want to remove `EXECUTE` privilege, replace `READ` with `EXECUTE`.
+
+[.wrap,gsql]
+----
+GSQL > REVOKE ACL PRIVILEGE READ ON QUERY example_query FROM role1 SECURED BY
+"owner_acl_password"
+----
+
+[CAUTION]
+====
+Removing `role1` from the ACL entry for `READ` privilege on `example_query` does not always mean that `role1` will not be able to access `example_query`.
+
+* If after removing `role1`, the ACL entry for `READ` privilege becomes empty, then RBAC governs access to the query.
```

Comment:
If you revoke acl privilege until the acl list becomes empty, the ACL entry will become "Nobody".
That's for safety so that ACL control is not lost if user didn't intend to do so. User will need to run "REVOKE ACL PRIVILEGE ... FROM ALL/*" to make it "Unspecified".